queries/atm.py

```diff
@@ -0,0 +1,571 @@
+        return []
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_deposit(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that accepts deposits."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("deposit", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_foreign_exchange(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that accepts foreign exchange."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("foreign_exchange", {}).get("active", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_coinmachine(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that has a coinmachine."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("coinmachine", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _format_voice_street_number(s: str) -> str:
+    """
+    Format street name for voice output,
+    also if it contains a range of numbers
+    for example: Fiskislóð 7-9 becomes Fiskislóð sjö til níu
+    """
+    # Formatting numbers
+    word_list = s.split()
+    last_word = word_list[-1]
+    if last_word[-1].isnumeric():
+        if "–" in last_word:
+            # replace irregular hyphen with regular hyphen
+            last_word.replace("–", "-")
+        if "-" in last_word:
+            split_nr = last_word.split("-")
+            first_number = number_to_text(split_nr[0], case="þf")
+            last_number = number_to_text(split_nr[-1], case="þf")
+            number = first_number + " til " + last_number
+            return " ".join(word_list[:-1]) + " " + number
+        return " ".join(word_list[:-1]) + " " + number_to_text(last_word, case="þf")
+    return s
```

Comment:
Ah, I think I forgot to mention the code we have for transcribing stuff for TTS in `speech/trans`, like the `gssml` function (you can see usage of it in e.g. the arithmetic module), but I notice now that we haven't implemented a method for transcribing addresses yet. This looks good for the time being though :).